lib/modules/apostrophe-express/index.js

```diff
@@ -500,14 +495,20 @@ module.exports = {
       }
       // Safe request establishes XSRF-TOKEN in session if not set already
       if ((req.method === 'GET') || (req.method === 'HEAD') || (req.method === 'TRACE')) {
+
         token = req.session && req.session['XSRF-TOKEN'];
+
         if (!token) {
-          token = self.apos.utils.generateId();
-          req.session['XSRF-TOKEN'] = token;
+          if (!req.user && self.options.session.disableAnonCsrf) {
+            token = 'csrf-fallback';
+          } else {
+            token = self.apos.utils.generateId();
+            req.session['XSRF-TOKEN'] = token;
+            // Reset the cookie so that if its lifetime somehow detaches from
+            // that of the session cookie we're still OK
+            res.cookie(self.apos.csrfCookieName, token, (self.options.csrf && self.options.csrf.cookie) || {});
+          }
         }
-        // Reset the cookie so that if its lifetime somehow detaches from
-        // that of the session cookie we're still OK
-        res.cookie(self.apos.csrfCookieName, token, (self.options.csrf && self.options.csrf.cookie) || {});
       } else {
         // All non-safe requests must be preceded by a safe request that establishes
         // the CSRF token, both as a cookie and in the session. Otherwise a user who is logged
```

Comment:
I don't see code here to check for the fallback token value in the appropriate situation. As it is I'm pretty sure this will reject any POST requests from a logged out user if disableAnonCsrf is set, because the browser side code has been updated to send "csrf-fallback" but the server side code has not been updated to accept that value.

Reply:
Please make sure a POST request from jQuery can succeed for both logged in and logged out users with the change in place and the option activated. Logged in is easy (opening any manage dialog is sufficient proof). Logged out... hmm. You might have to create a test route and hit it from jquery. Any $.post or $.ajax or $.jsonCall will do (GET is not checked, by design).